org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/CompositeADTabbox.java

```diff
@@ -1067,7 +1067,7 @@ private void onActivateDetail(IADTabpanel tabPanel) {
 			if (!tabPanel.getGridTab().isSortTab()) {
 				currentRow = tabPanel.getGridTab().getCurrentRow();
 			}
-			tabPanel.query(false, 0, 0);			
+			tabPanel.query(false, 0, tabPanel.getGridTab().getMaxQueryRecords());
```

Comment:
I did a workspace reference search for "IADTabpanel.query(boolean currentRows, int currentDays, int maxRows)" and it seems we are doing this consistently (using GridTab.getMaxQueryRecords()). I think those call needs to be fixed as well.

Reply:
Thanks @hengsin - I just found one more - fixed

Reply:
AbstractADWindowContext
- 651 inside doZoomToDetail
WQuickForm
- 460, inside dispose

Reply:
> AbstractADWindowContext
>     * 651 inside doZoomToDetail
> WQuickForm
>     * 460, inside dispose

Those were changed on the first commit.